### PR TITLE
wolfram-engine: livecheck and download url update

### PR DESCRIPTION
**Pull request by daeho-ro**
Update download and livecheck url, because the previous livecheck is not updated automatically.
Download url is now same as the web page.

---

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [] `brew style --fix <cask>` reports no offenses.

---

### CODE REVIEW

File: Casks/w/wolfram-engine.rb [hunk 1]
```diff
@@ -2,15 +2,16 @@
   version "14.0.0.0"
   sha256 "6f281cc645d63e98180327f084fda3becd78ae5c977e2026f71ed48e2999d381"
 
-  url "https://files.wolframcdn.com/packages/Homebrew/#{version}/WolframEngine_#{version.major_minor_patch}_MAC.dmg",
+  url "https://files.wolframcdn.com/WolframEngine/#{version}/WolframEngine_#{version.major_minor_patch}_MAC_DLM.dmg",
       verified: "files.wolframcdn.com/packages/Homebrew/"
```
Comment (SMillerDev) [inline Casks/w/wolfram-engine.rb:6]:
```suggestion
      verified: "files.wolframcdn.com/WolframEngine/"
```

File: Casks/w/wolfram-engine.rb [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
   name "Wolfram Engine"
   desc "Evaluator for the Wolfram Language"
   homepage "https://www.wolfram.com/engine/"
 
   livecheck do
-    url "https://files.wolframcdn.com/packages/Homebrew/latest.txt"
-    regex(/^(\d+(?:\.\d+)+)$/i)
+    url "https://account.wolfram.com/dl/WolframEngine?platform=Mac"
+    regex(/WolframEngine\/(\d+(?:\.\d+)+)/i)
```
Comment (SMillerDev) [inline Casks/w/wolfram-engine.rb:13]:
```suggestion
    regex(%r{WolframEngine\/(\d+(?:\.\d+)+)}i)
```

**Comment by daeho-ro** [comment]
Oops, the package is not signed.

```
 - Signature verification failed:
/private/tmp/cask-audit20240823-62668-3dqp5p/Wolfram Engine.app: invalid API object reference

macOS on ARM requires software to be signed.
Please contact the upstream developer to let them know they should sign and notarize their software.
```

Maybe this is the reason to split the `.dmg` for homebrew. so sad.

**Comment by daeho-ro** [comment]
Hold the download url, and only changes the livecheck url.

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**